src/sleep/sleep.c

```diff
@@ -44,49 +48,107 @@
 
 static SleepOperation arg_operation = _SLEEP_OPERATION_INVALID;
 
-static int write_hibernate_location_info(const HibernateLocation *hibernate_location) {
+static int write_efi_hibernate_info(const char *device, const char *offset, int log_level) {
+        int r = 0;
+
+#if ENABLE_EFI_HIBERNATE_INFO
+        _cleanup_(blkid_free_probep) blkid_probe b = NULL;
+        _cleanup_free_ char *info = NULL;
+        char info_name[STRLEN("HibernateInfo-") + SD_ID128_STRING_MAX + STRLEN("-" EFI_VENDOR_SYSTEMD_STR)];
+        const char *uuid;
+        sd_id128_t machine_id = SD_ID128_NULL;
+
+        if (!is_efi_boot())
+                return 0;
+
+        errno = 0;
+        b = blkid_new_probe_from_filename(device);
```

Comment:
hmm, use use blkid here, any specific reason? udev runs at this point already, so we can rely on the data udev already probed for us. that should be faster and simpler